src/details/hooks.cpp

```diff
@@ -0,0 +1,19 @@
+#include "details/hooks.h"
+
+namespace kaleidoscope {
+
+__attribute__((weak))
```

Comment:
Added weak implementation of `Hooks`'s static methods to enable backwards compatibility. Without this, any sketch that does not register new plugins with `KALEIDOSCOPE_INIT_PLUGINS(...)` would cause undefined symbol errors because of the missing methods of class `Hooks`.